coding/src/analysis.py

```diff
@@ -45,7 +46,33 @@ def __init__(self, results_path, ends_with='.pickle', matching_strategy='substr'
         self.populate_category_probs()
         # populate scores
         self.populate_scores()
-
+        #populate margin
+        self.populate_margin()
+    
+    def populate_margin(self):
+        '''Finds the margin between the probability of the correct category
+        and the probability of the next category of highest weight'''
+        
+        picklepath = 'experiments/nyt/07-27-2021/ambiguity/ambiguitycandidates/ambiguity_candidates_w_margin.pickle'
```

Comment:
Is there a way we could make this an argument? If we wanted to do it for a different set of candidates that would be nice